app/projects/admin.py

```diff
@@ -0,0 +1,28 @@
+from django.contrib import admin
+from app.projects.models import Project
+from flask import current_app
+
+
+class ProjectAdmin(admin.ModelAdmin):
+    ordering = ["-created_at"]
+    readonly_fields = ["created_by", "uuid"]
+    fields = ["uuid", "created_by", "name", "domain", "retention_period", "owned_by"]
+    list_display = ["name", "owned_by", "created_at", "retention_period", "uuid"]
+    list_filter = ["retention_period", "owned_by"]
+    search_fields = ["name", "uuid"]
+
+    # Only show the created_by field on existing projects
+    def get_fields(self, request, obj=None):
+        fields = list(super().get_fields(request, obj=obj))
+        if obj is None:
+            fields.remove("uuid")
+            fields.remove("created_by")
```

Comment:
What happens on the create view? Is this like strictly necessary? I think this is fine. Let's keep Django admin as simple as possible code-wise, unless we commit to it.

Let's just be mindful of not improving the user experience too much.

Reply:
On create view, without this it shows a random UUID (not the one which is actually going to be used) and a blank "created by" field. Both of which I think would be confusing for users.

Reply:
Oh! Even with editable=False?

Reply:
This makes total sense, forget me.